src/megaclient.cpp

```diff
@@ -11889,14 +11812,7 @@ void MegaClient::fetchnodes(bool nocache)
         scsn.clear();
 
 #ifdef ENABLE_SYNC
-        // lets remove the active syncs. There could be some when enforcing fetchnodes(true)
-        // after API_ETOOMANY (too many action packets)
-        for (sync_list::iterator it = syncs.begin(); it != syncs.end(); it++)
-        {
-            (*it)->changestate(SYNC_CANCELED);//we set it as cancelled. It will be removed from active sync latter on
-            // Note: this does not cause the sync to be removed from syncMap.
-            // However, after processing the new FetchNodes, they will be resumed, overriding the existing ones in the map.
-        }
+        assert(!syncs.hasRunningSyncs()); // not loaded yet - deferred to fetchnodes reply
```

Comment:
This assert fails, @mattw-mega (try to `sync` a folder with megacli and run `reload`).
```
MEGAcli: ../../../src/megaclient.cpp:11815: void mega::MegaClient::fetchnodes(bool): Assertion `!syncs.hasRunningSyncs()' failed.
```
The comment you removed still applies:
>  // lets remove the active syncs. There could be some when enforcing fetchnodes(true)

Could we simply keep the existing behavior? or would it be problematic?

Reply:
This is a tricky one - really, reloading like this is not supported by the current sync algorithm, and should be impossible for MEGAsync.   Given that we will execute fetchnodes and get the last cached tree (not matching our current SCSN, most likely), and then receive actionpackets from there, some of which were already applied to the sync and some not - there's really no way to get this right (until the sync rework branch is ready).   I've opted to disable any active syncs at this point, with an explanatory new error code.

Reply:
Yes, I think that disabling syncs is the right (conservative) approach in this case, though we may get an increase of user complains that get their syncs disabled upon `API_ETOOMANY` scenarios. Better safe than sorry